src/organizations/components/OrgProfileTab/index.tsx

```diff
@@ -1,64 +1,74 @@
 // Libraries
-import React, {FC, useEffect} from 'react'
-import {useSelector, useDispatch} from 'react-redux'
+import React, {FC, useEffect, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
 
 // Components
 import {
-  FlexBox,
   AlignItems,
-  FlexDirection,
   ComponentSize,
+  FlexBox,
+  FlexDirection,
   JustifyContent,
 } from '@influxdata/clockface'
-import UsersProvider from 'src/users/context/users'
 import LabeledData from 'src/organizations/components/OrgProfileTab/LabeledData'
 import CopyableLabeledData from 'src/organizations/components/OrgProfileTab/CopyableLabeledData'
-import DeletePanel from 'src/organizations/components/OrgProfileTab/DeletePanel'
+import {DeletePanel} from 'src/organizations/components/OrgProfileTab/DeletePanel'
 
-// Types
-import {RemoteDataState} from 'src/types'
+// Notifications
+import {orgDetailsFetchError} from 'src/shared/copy/notifications'
+import {notify} from 'src/shared/actions/notifications'
 
-// Utils
-import {CLOUD} from 'src/shared/constants'
+// Providers
+import UsersProvider from 'src/users/context/users'
 
 // Selectors
 import {getMe} from 'src/me/selectors'
 import {getOrg} from 'src/organizations/selectors'
-import {
-  selectCurrentIdentity,
-  selectQuartzOrgDetailsStatus,
-} from 'src/identity/selectors'
+import {selectCurrentOrg} from 'src/identity/selectors'
 
 // Thunks
 import {getCurrentOrgDetailsThunk} from 'src/identity/actions/thunks'
 
+// Types
+import {RemoteDataState} from 'src/types'
+
+// Constants
+import {CLOUD} from 'src/shared/constants'
+
 // Styles
 import 'src/organizations/components/OrgProfileTab/style.scss'
 
 const OrgProfileTab: FC = () => {
   const me = useSelector(getMe)
   const org = useSelector(getOrg)
-  const {org: quartzOrg} = useSelector(selectCurrentIdentity)
-  const orgDetailsStatus = useSelector(selectQuartzOrgDetailsStatus)
-
+  const quartzOrg = useSelector(selectCurrentOrg)
   const dispatch = useDispatch()
 
-  const identityOrgId = quartzOrg.id
+  // Data about the user's organization is intentionally re-fetched when this component mounts again.
+  const [orgDetailsStatus, setOrgDetailsStatus] = useState(
+    RemoteDataState.NotStarted
+  )
+
+  const orgDetailsLoaded = orgDetailsStatus === RemoteDataState.Done
 
   useEffect(() => {
     if (!CLOUD) {
       return
     }
 
     if (orgDetailsStatus === RemoteDataState.NotStarted) {
-      dispatch(getCurrentOrgDetailsThunk(identityOrgId))
+      Promise.resolve(dispatch(getCurrentOrgDetailsThunk(quartzOrg.id)))
+        .then(() => {
+          setOrgDetailsStatus(RemoteDataState.Done)
+        })
+        .catch(() => {
+          setOrgDetailsStatus(RemoteDataState.Error)
+          dispatch(notify(orgDetailsFetchError()))
+        })
```

Comment:
I'm a little confused about this usage of `Promise.resolve`. Typically `Promise.resolve` is used to convert an existing value into a promise that is then returned from a method that needs to be async. What's the intent here? Is it to handle an error thrown from `getCurrentOrgDetailsThunk` ?

I think `dispatch` is a synchronous function, so can you do somethign like

```ts
try {
  dispatch(getCurrentOrgDetailsThunk(quartzOrg.id))
  setOrgDetailsStatus(RemoteDataState.Done)
} catch {
  setOrgDetailsStatus(RemoteDataState.Error)
  dispatch(notify(orgDetailsFetchError()))
}
```

Reply:
The issue here is that, because dispatch is synchronous but the thunk is asynchronous, that code will never hit the catch block whether there is an error or not (which is obnoxious). 😠 

On testing, the way this works is: the thunk is dispatched, `setOrgDetailsStatus` is set to `RemoteDataState.Done`, we exit the `try/catch` block, and then the thunk (at a later point, whenever we get a bad response from the API) will throw an uncaught error. So, we either need a promise chain or an await, and Typescript doesn't accept `await` as valid since `dispatch` doesn't return a promise.

I agree with you it's better to use a `try/catch` here. So here's a solution that hews a little closer to your suggestion - we just need to add an `await`, and also a `Promise.resolve` to convert the return value from the dispatch into a function.

```
  useEffect(() => {
    if (!CLOUD) {
      return
    }

    const retrieveOrgDetails = async () => {
      if (orgDetailsStatus === RemoteDataState.NotStarted) {
        try {
          await Promise.resolve(
            dispatch(getCurrentOrgDetailsThunk(quartzOrg.id))
          )
          setOrgDetailsStatus(RemoteDataState.Done)
        } catch (err) {
          setOrgDetailsStatus(RemoteDataState.Error)
          dispatch(notify(orgDetailsFetchError()))
        }
      }
    }

    retrieveOrgDetails()
  }, [dispatch, orgDetailsStatus, quartzOrg.id])
```